src/components/Common/ButtonWithDropdown/index.tsx

```diff
@@ -39,21 +38,37 @@ const DropdownItem: React.FC<DropdownItemProps> = ({
   );
 };
 
-interface ButtonWithDropdownProps
-  extends ButtonHTMLAttributes<HTMLButtonElement> {
+// Helper type to override types (overrides onClick)
+type MergeElementProps<
+  T extends React.ElementType,
+  P extends Record<string, unknown>
+> = Omit<React.ComponentProps<T>, keyof P> & P;
+
+type ElementTypes = 'button' | 'a';
+
+type ButtonWithDropdownBaseProps<P> = {
   text: ReactNode;
   dropdownIcon?: ReactNode;
   buttonType?: 'primary' | 'ghost';
-}
+  // Had to do declare this manually as typescript would assume e was of type any otherwise
+  onClick?: (
+    e: React.MouseEvent<P extends 'a' ? HTMLAnchorElement : HTMLButtonElement>
+  ) => void;
+};
+
+type ButtonWithDropdownProps<P extends React.ElementType> = {
+  as?: P;
+} & MergeElementProps<P, ButtonWithDropdownBaseProps<P>>;
 
-const ButtonWithDropdown: React.FC<ButtonWithDropdownProps> = ({
+function ButtonWithDropdown<P extends ElementTypes = 'button'>({
+  as,
   text,
   children,
   dropdownIcon,
   className,
   buttonType = 'primary',
   ...props
-}) => {
+}: ButtonWithDropdownProps<P>): JSX.Element {
```

Comment:
This would mean that a dropdown could only have one type, as in only buttons, or only anchors. I'm not sure we want that? I haven't thought it through.

Reply:
The dropdown is actually only anchors already.  Only the button itself can be one or the other...